cmd/cluster/transferowner.go

```diff
@@ -54,9 +66,175 @@ func newTransferOwnerOptions(streams genericclioptions.IOStreams, globalOpts *gl
+		Template:       template,
+		ClusterId:      clusterId,
+		TemplateParams: []string{fmt.Sprintf("DATE=%v", time.Now().UTC().Format(time.RFC3339))},
+	}
+}
+
+func updatePullSecret(conn *sdk.Connection, kubeCli client.Client, clientset *kubernetes.Clientset, clusterID string, pullsecret []byte) error {
+	currentEnv := utils.GetCurrentOCMEnv(conn)
+	secretName := "pull"
+	hiveNamespace := "uhc-" + currentEnv + "-" + clusterID
+
+	clusterDeployments := &hiveapiv1.ClusterDeploymentList{}
+	if err := kubeCli.List(context.TODO(), clusterDeployments, client.InNamespace(hiveNamespace)); err != nil {
+		return err
+	}
+
+	if len(clusterDeployments.Items) == 0 {
+		return fmt.Errorf("failed to retreive cluster deployments")
+	}
+	cdName := clusterDeployments.Items[0].ObjectMeta.Name
+
+	// Delete the secret
+	err := clientset.CoreV1().Secrets(hiveNamespace).Delete(context.TODO(), secretName, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: hiveNamespace,
+		},
+		Type: corev1.SecretTypeDockerConfigJson,
+		Data: map[string][]byte{
+			".dockerconfigjson": pullsecret,
+		},
+	}
+	_, err = clientset.CoreV1().Secrets(hiveNamespace).Create(context.TODO(), secret, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	//Create a syncset on hive to push the pull secret down
+	syncsetName := "pull-secret-replacement"
+
+	err = createSyncSet(syncsetName, hiveNamespace, cdName, kubeCli)
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+func createSyncSet(syncSetName string, hiveNamespace string, cdName string, kubeCli client.Client) error {
+	ctx := context.TODO()
+
+	syncSet := &hiveapiv1.SyncSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      syncSetName,
+			Namespace: hiveNamespace,
+		},
+		Spec: hiveapiv1.SyncSetSpec{
+			ClusterDeploymentRefs: []corev1.LocalObjectReference{
+				{
+					Name: cdName,
+				},
+			},
+			SyncSetCommonSpec: hiveapiv1.SyncSetCommonSpec{
+				ResourceApplyMode: "Upsert",
+				Secrets: []hiveapiv1.SecretMapping{
+					{
+						SourceRef: hiveapiv1.SecretReference{
+							Name:      "pull",
+							Namespace: hiveNamespace,
+						},
+						TargetRef: hiveapiv1.SecretReference{
+							Name:      "pull-secret",
+							Namespace: "openshift-config",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	err := kubeCli.Create(ctx, syncSet)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("SyncSet %s in namespace %s has been created.\n", syncSetName, hiveNamespace)
+
+	hiveinternalv1alpha1.AddToScheme(kubeCli.Scheme())
+	searchStatus := &hiveinternalv1alpha1.ClusterSync{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cdName,
+			Namespace: hiveNamespace,
+		},
+	}
+	foundStatus := &hiveinternalv1alpha1.ClusterSync{}
+	isSSSynced := false
+	for i := 0; i < 24; i++ {
+		err = kubeCli.Get(ctx, client.ObjectKeyFromObject(searchStatus), foundStatus)
+		if err != nil {
+			return err
```

Comment:
Same as the few above
```suggestion
		return fmt.Errorf("failed to delete syncset from hive: %w", err)
```